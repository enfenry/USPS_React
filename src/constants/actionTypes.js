

//<--------------------- APPLICATIONS ------------------------>
export const READ_APPLICATIONS_SUCCESSFUL = 'READ_APPLICATIONS_SUCCESSFUL';
export const READ_APPLICATIONS_PENDING = 'READ_APPLICATIONS_PENDING';
export const READ_APPLICATIONS_FAILURE = 'READ_APPLICATIONS_FAILURE';

export const CREATE_APPLICATION_SUCCESSFUL = 'CREATE_APPLICATION_SUCCESSFUL';
export const CREATE_APPLICATION_FAILURE = 'CREATE_APPLICATION_FAILURE';

export const UPDATE_APPLICATION_SUCCESSFUL = 'UPDATE_APPLICATION_SUCCESSFUL';
export const UPDATE_APPLICATION_FAILURE = 'UPDATE_APPLICATION_FAILURE';

export const DELETE_APPLICATION_SUCCESSFUL = 'DELETE_APPLICATION_SUCCESSFUL';
export const DELETE_APPLICATION_FAILURE = 'DELETE_APPLICATION_FAILURE';

//<--------------------- CUSTOMERS ------------------------>
export const READ_CUSTOMERS_SUCCESSFUL = 'READ_CUSTOMERS_SUCCESSFUL';
export const READ_CUSTOMERS_PENDING = 'READ_CUSTOMERS_PENDING';
export const READ_CUSTOMERS_FAILURE = 'READ_CUSTOMERS_FAILURE';

export const CREATE_CUSTOMER_SUCCESSFUL = 'CREATE_CUSTOMER_SUCCESSFUL';
export const CREATE_CUSTOMER_FAILURE = 'CREATE_CUSTOMER_FAILURE';

export const UPDATE_CUSTOMER_SUCCESSFUL = 'UPDATE_CUSTOMER_SUCCESSFUL';
export const UPDATE_CUSTOMER_FAILURE = 'UPDATE_CUSTOMER_FAILURE';

export const DELETE_CUSTOMER_SUCCESSFUL = 'DELETE_CUSTOMER_SUCCESSFUL';
export const DELETE_CUSTOMER_FAILURE = 'DELETE_CUSTOMER_FAILURE';

//<--------------------- ORDERS ------------------------>
export const READ_ORDERS_SUCCESSFUL = 'READ_ORDERS_SUCCESSFUL';
export const READ_ORDERS_PENDING = 'READ_ORDERS_PENDING';
export const READ_ORDERS_FAILURE = 'READ_ORDERS_FAILURE';

//<--------------------- PRODUCTS ------------------------>
export const READ_PRODUCTS_SUCCESSFUL = 'READ_PRODUCTS_SUCCESSFUL';
export const READ_PRODUCTS_PENDING = 'READ_PRODUCTS_PENDING';
export const READ_PRODUCTS_FAILURE = 'READ_PRODUCTS_FAILURE';

//<---------------------ADDRESSES ------------------------>
export const READ_ADDRESSES_SUCCESSFUL = 'READ_ADDRESSES_SUCCESSFUL';
export const READ_ADDRESSES_PENDING = 'READ_ADDRESSES_PENDING';
export const READ_ADDRESSES_FAILURE = 'READ_ADDRESSES_FAILURE';
