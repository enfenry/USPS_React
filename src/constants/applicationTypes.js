export const ADDRESS_CHANGE = 717800000;
export const MAIL_FORWARDING = 717800001;
export const PACKAGE_SUBMISSION = 717800002;