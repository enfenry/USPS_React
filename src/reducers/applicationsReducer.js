import { CREATE_APPLICATION_SUCCESSFUL, CREATE_APPLICATION_FAILURE, READ_APPLICATIONS_SUCCESSFUL, READ_APPLICATIONS_PENDING, READ_APPLICATIONS_FAILURE,
         UPDATE_APPLICATION_SUCCESSFUL, UPDATE_APPLICATION_FAILURE, DELETE_APPLICATION_SUCCESSFUL, DELETE_APPLICATION_FAILURE, 
         APP_TO_ORDER_SUCCESSFUL, APP_TO_ORDER_FAILURE} from '../constants/actionTypes'
export default function applicationsReducer(state = {}, action) {

    switch (action.type) {

        case CREATE_APPLICATION_SUCCESSFUL: {
        let newApplications = [...state.applications];
        newApplications.push(action.data);
            return { ...state, applications: newApplications, requestState: {applicationsCreateSuccess: true, applicationsCreateFailed: false, error: null }}; 
        }
        case CREATE_APPLICATION_FAILURE:
            return { ...state, applications:state.applications, requestState: {applicationsCreateSuccess: false, applicationsCreateFailed: true, error: action.data }};    

        case READ_APPLICATIONS_SUCCESSFUL:
            return { ...state, applications: action.data.value, requestState: { applicationsReadSuccess: true, applicationsReadPending: false, applicationsReadFailed: false, error: null }};
        case READ_APPLICATIONS_PENDING:
            return { ...state, requestState: { applicationsReadSuccess: false, applicationsReadPending: true, applicationsReadFailed: false, error: null  }};
        case READ_APPLICATIONS_FAILURE:
            return { ...state, requestState: { applicationsReadSuccess: false, applicationsReadPending: false, applicationsReadFailed: true, error: action.error}};

        case UPDATE_APPLICATION_SUCCESSFUL: {
            const itemIndex = state.applications.findIndex((e) => (e.ss_applicationid === action.id));
            let apps = [...state.applications];   
            apps[itemIndex]=action.data;
            return { ...state, applications: apps, requestState: {applicationsUpdateSuccess: true,  applicationsUpdateFailed: false, error: null  }};
        }
        case UPDATE_APPLICATION_FAILURE:
            return { ...state, applications:state.applications, requestState: {applicationsUpdateSuccess: false, applicationsUpdateFailed: true, error: action.error}};    

        case DELETE_APPLICATION_SUCCESSFUL:
            return { ...state, applications: state.applications.filter((e) => e.ss_applicationid !== action.data), requestState: {applicationsDeleteSuccess: true, applicationsDeleteFailed: false, error: null  }};
        case DELETE_APPLICATION_FAILURE:
            return { ...state, applicationsDeleteSuccess: false,  requestState: { applicationsDeleteFailed: true, error: action.error }};

        case APP_TO_ORDER_SUCCESSFUL:
            return { ...state, requestState: { applicationsToOrderSuccess: true,  applicationsToOrderFailed: false, error: null }};
        case APP_TO_ORDER_FAILURE:
            return { ...state, requestState: { applicationsToOrderSuccess: false, applicationsToOrderFailed: true, error: action.error }};     

        default:
            return state;
    }
}