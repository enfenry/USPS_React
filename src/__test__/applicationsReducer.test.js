import applicationsReducer from '../reducers/applicationsReducer';
import { READ_APPLICATIONS_SUCCESSFUL, READ_APPLICATIONS_PENDING, READ_APPLICATIONS_FAILURE, DELETE_APPLICATION_SUCCESSFUL } from '../constants/actionTypes';
import expect from 'expect';

describe('applications reducer', () => {

    it('should return the initial state', () => {
        expect(applicationsReducer(undefined, {
            type: "DUMMY_TYPE"
        })).toMatchSnapshot();
    });

    it('should handle READ_APPLICATIONS_PENDING', () => {
        expect(applicationsReducer({}, {
            type: READ_APPLICATIONS_PENDING
        })).toMatchSnapshot();
    });

    it('should handle READ_APPLICATIONS_FAILURE', () => {
        expect(applicationsReducer({}, {
            type: READ_APPLICATIONS_FAILURE
        })).toMatchSnapshot();
    });

    it('should handle READ_APPLICATIONS_SUCCESFUL', () => {
        expect(applicationsReducer({}, {
            type: READ_APPLICATIONS_SUCCESSFUL,
            data: {
                value: [
                    "dummy data"
                ]
            }
        })).toMatchSnapshot();
    });

    it('should handle DELETE_APPLICATION_SUCCESSFUL', () => {
        expect(
            applicationsReducer(
                {
                    applications: [
                        { ss_applicationid: "a" },
                        { ss_applicationid: "b" },
                        { ss_applicationid: "c" }
                    ]
                },
                {
                    type: DELETE_APPLICATION_SUCCESSFUL,
                    data: "b"
                }
            )
        ).toMatchSnapshot();
    });
});