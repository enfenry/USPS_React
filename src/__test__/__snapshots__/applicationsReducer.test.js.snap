// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications reducer should handle DELETE_APPLICATION_SUCCESSFUL 1`] = `
Object {
  "applications": Array [
    Object {
      "ss_applicationid": "a",
    },
    Object {
      "ss_applicationid": "c",
    },
  ],
  "applicationsRequestFailed": false,
  "applicationsRequestPending": false,
  "applicationsRequestSuccess": true,
}
`;

exports[`applications reducer should handle READ_APPLICATIONS_FAILURE 1`] = `
Object {
  "applicationsRequestFailed": true,
  "applicationsRequestPending": false,
  "applicationsRequestSuccess": false,
}
`;

exports[`applications reducer should handle READ_APPLICATIONS_PENDING 1`] = `
Object {
  "applicationsRequestFailed": false,
  "applicationsRequestPending": true,
  "applicationsRequestSuccess": false,
}
`;

exports[`applications reducer should handle READ_APPLICATIONS_SUCCESFUL 1`] = `
Object {
  "applications": Array [
    "dummy data",
  ],
  "applicationsRequestFailed": false,
  "applicationsRequestPending": false,
  "applicationsRequestSuccess": true,
}
`;

exports[`applications reducer should return the initial state 1`] = `Object {}`;
